source/NationalInstruments/jquery.flot.annotations.js

```diff
@@ -0,0 +1,394 @@
+            for (let i = 0; i < lines.length; i++) {
+                let size = ctx.measureText(lines[i]);
+                if (size.width > longest.width) {
+                    longest = size;
+                }
+            }
+
+            return longest;
+        }
+
+        lineHeight (plot) {
+            let surface = plot.getSurface();
+            let styles = window.getComputedStyle(surface.element);
+            let lineHeight = parseFloat(styles['lineHeight']);
+            return lineHeight;
+        }
+
+        _drawBox (plot, ctx, annotations, content, formattedText, offset) {
+            let x = content.x;
+            let y = content.y;
+            let xaxis = plot.getXAxes()[0];
+            let yaxis = plot.getYAxes()[0];
+            let arrowDirn = content.arrowDirection;
+            let lineCount = this.lineCount(formattedText);
+            if (annotations.location === 'absolute') {
+                x = xaxis.p2c(content.x);
+                y = yaxis.p2c(content.y);
+            } else {
+                x = content.x * plot.width();
+                y =content.y * plot.height();
+            }
+
+            let padding = annotations.padding;
+            x += offset.x;
+            y += offset.y;
+            let size = this.lineWidth(ctx, formattedText);
+            let width = size.width + 2 * padding;
+            let height = lineCount * this._lineHeight + 2 * padding;
+
+            if (arrowDirn === '') {
+                arrowDirn = this.defaultArrowDirection(plot, x, y);
+            }
+
+            switch (arrowDirn) {
+                case 'n':
+                    x = x - width / 2;
+                    y = y - height;
+                    break;
+                case 's':
+                    x = x - width / 2;
+                    break;
+                case 'e':
+                    x = x - width;
+                    y = y - height / 2;
+                    break;
+                case 'w':
+                    y = y - height / 2;
+                    break;
+                case 'ne':
+                    y = y - height;
+                    break;
+                case 'nw':
+                    x = x - width;
+                    y = y - height;
+                    break;
+                case 'se':
+                    break;
+                case 'sw':
+                    x = x - width;
+                    break;
+            }
+
+            ctx.save();
+            ctx.strokeStyle = annotations.borderColor;
+            ctx.fillStyle = annotations.backgroundColor;
+            ctx.lineWidth = annotations.borderThickness;
+            ctx.beginPath();
+            ctx.fillRect(x, y, width, height);
+            ctx.strokeRect(x, y, width, height);
+            ctx.restore();
+
+            return {x: x, y: y, width: width, height: height};
+        }
+
+        _drawContent (plot, ctx, annotations, formattedText, bounds) {
+            let lines = formattedText.split(/\r\n|\r|\n/);
```

Comment:
Can you leverage the text drawing code in jquery.canvaswrapper.js here?

Reply:
jquery.canvaswrapper.js uses `<br>` tags for line breaks instead of newline chars. We should probably standardize on using `<br>`, or update the existing code to work with newline chars as well.

Reply:
So most of the plugins use fillText not canvaswrapper. canvaswrapper seems to be used mostly in the axes for tick labels and axishandles. Unless there is more consensus to go with canvaswrapper I am going to stick with fillText.

However I did change '\n' to '<br>'